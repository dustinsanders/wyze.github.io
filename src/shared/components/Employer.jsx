// @flow

import type { Style, VNode } from '../../types'
import { after, before, css, media, nthChild } from 'glamor'
import { large, small } from '../styles/breakpoints'
import { level } from '../styles'
import H2 from './H2'
import H3 from './H3'
import Section from './Section'
import Strong from './Strong'
import preact from 'preact'

type Props = {
  end?: string,
  name: string,
  start: string,
}

const section: Style = css(
  {
    paddingBottom: '1em',
    width: '100%',
  },
  media(small, {
    paddingBottom: '.5em',
  }),
  media(large, css(
    {
      width: '50%',
    },
    nthChild('even', {
      paddingLeft: '.5em',
    }),
    nthChild('odd', {
      paddingRight: '.5em',
    }),
  )),
)
const child: Style = css(
  {
    display: 'block',
  },
  media(small, {
    alignItems: 'center',
    display: 'flex',
    flex: '0 0 50%',
    justifyContent: 'space-between',
  }),
)
const h3: Style = css(
  {
    fontWeight: 300,
  },
  media(small,
    after({
      content: '")"',
    }),
    before({
      content: '"("',
    }),
  ),
)

const Employer = ({ end, name, start }: Props): VNode => (
  <Section styles={section}>
    <div {...css(level, child)}>
      <H2><Strong text={name} /></H2>
      <H3 styles={h3}>{start} &mdash; {end}</H3>
    </div>
  </Section>
)

Employer.defaultProps = {
  end: 'Present',
}

export default Employer
