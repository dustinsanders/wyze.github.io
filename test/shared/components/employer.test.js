// @flow

import Employer from '../../../src/shared/components/Employer'
import snapshot from '../../helpers/snapshot'

const present = {
  name: 'ABC Company',
  start: 'May 2016',
}

const past = {
  ...present,
  end: 'December 2016',
}

describe('<Employer />', () => {
  it('renders', snapshot.bind(null, Employer, past))
  it('renders without end', snapshot.bind(null, Employer, present))
})
