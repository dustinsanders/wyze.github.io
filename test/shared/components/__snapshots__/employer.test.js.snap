// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Employer /> renders 1`] = `
"<Section styles=\\"css-1sx3me2\\">
  <div data-css-13yqx3a>
    <H2>
      <Strong text=\\"ABC Company\\"></Strong>
    </H2>
    <H3 styles=\\"css-mqz3x4\\">May 2016 — December 2016</H3>
  </div>
</Section>"
`;

exports[`<Employer /> renders without end 1`] = `
"<Section styles=\\"css-1sx3me2\\">
  <div data-css-13yqx3a>
    <H2>
      <Strong text=\\"ABC Company\\"></Strong>
    </H2>
    <H3 styles=\\"css-mqz3x4\\">May 2016 — Present</H3>
  </div>
</Section>"
`;
